tests/utils_test.py

```diff
@@ -106,7 +108,10 @@ def sort_nested(data):
         new_data = {}
         for k, v in data.items():
             if isinstance(v, list):
-                v = sorted(v)
+                if isinstance(v[0], dict):
+                    v = sorted(v, key=lambda x: json.dumps(x, sort_keys=True))
+                else:
+                    v = sorted(v)
```

Comment:
I don't get the logic. Why don't we just call `sort_nested(v)` here?
Instead, the sorting used here should be applied to `data` in the case where it is a list